Support Iterable<Project> in Kotlin DSL dependencies block (#38023)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

### Build authoring improvements
Gradle provides [rich APIs](userguide/getting_started_dev.html) for build engineers and plugin authors, enabling the creation of custom, reusable build logic and better maintainability.

> #### Kotlin DSL: pass `Iterable<Project>` to a configuration

Reviewer: ```suggestion
#### Kotlin DSL support for adding multiple project dependencies
```

Author: Done

> The Kotlin DSL `dependencies { ... }` block now accepts an `Iterable<Project>` (e.g. `allprojects`, `subprojects`) and adds one project dependency per element, mirroring the existing Groovy DSL convenience. Previously the same call would treat the iterable as a single dependency notation and fail.

Reviewer: ```suggestion
The [Kotlin DSL](userguide/kotlin_dsl.html) `dependencies { ... }` block now accepts an [`Iterable<Project>`](javadoc/org/gradle/api/Project.html) (e.g. `allprojects`, `subprojects`) and adds one project dependency per element, mirroring the existing Groovy DSL convenience:
```

```kotlin
configurations.create("integrationApi")

dependencies {
    "integrationApi"(subprojects)   // one project dependency per subproject
}
```

> 

Reviewer: ```suggestion

Previously the same call would treat the iterable as a single dependency notation and fail.

```

The new overloads internally use the recommended `DependencyHandler.project(String)` API, so they remain valid after the deprecation of `Project`-as-dependency-notation is fully enforced.

> 

Reviewer: ```suggestion

See the [Configurations and dependencies](userguide/migrating_from_groovy_to_kotlin_dsl.html#configurations-and-dependencies) section in the Gradle User Manual for more details.

```
